packages/xo-web/src/xo-app/index.js

```diff
@@ -234,6 +241,87 @@ export const ICON_POOL_LICENSE = {
     isXoaStatusOk: ({ xoaStatus }) => !xoaStatus.includes('✖'),
     areHostsVersionsEqualByPool: ({ hostsByPoolId }) =>
       mapValues(hostsByPoolId, hosts => every(hosts, host => host.version === hosts[0].version)),
+    xostorLicenseInfoByXostorId: (
+      { xcpngLicenseByBoundObjectId, xostorLicensesByBoundObjectId, hostsByPoolId },
+      { xostors }
+    ) => {
+      if (xcpngLicenseByBoundObjectId === undefined || xostorLicensesByBoundObjectId === undefined) {
+        return 
+      }
+      const xostorLicenseInfoByXostorId = {}
+      const now = Date.now()
+
+      forEach(xostors, xostor => {
+        const xostorId = xostor.id
+        const hosts = hostsByPoolId[xostor.$pool]
+
+        const alerts = []
+        let supportEnabled = true
+
+        hosts.forEach(host => {
+          const hostId = host.id
+          const xcpngLicense = xcpngLicenseByBoundObjectId[hostId]
+          const xostorLicenses = xostorLicensesByBoundObjectId[hostId]
+
+          if (xcpngLicense === undefined || xcpngLicense.expires < now) {
+            supportEnabled = false
+            alerts.push({
+              level: 'danger',
+              render: (
+                <p>
+                  {_('hostNoSupport')} <RenderXoHost id={hostId} />
+                </p>
+              ),
+            })
+          }
+
+          if (xostorLicenses === undefined) {
+            supportEnabled = false
+            alerts.push({
+              level: 'danger',
+              render: <p>{_('hostHasNoXostorLicense', { host: <RenderXoHost id={hostId} /> })}</p>,
+            })
+          }
+
+          if (xostorLicenses?.length > 1) {
+            alerts.push({
+              level: 'warning',
+              render: (
+                <p>
+                  {_('hostBoundToMultipleXostorLicenses', { host: <RenderXoHost id={hostId} /> })}
+                  <br />
+                  {xostorLicenses.map(license => license.id.slice(-4)).join(',')}
+                </p>
+              ),
+            })
+          }
+
+          const expiredXostorLicenses = xostorLicenses?.filter(license => license.expires < now)
+          if (expiredXostorLicenses?.length > 0) {
+            supportEnabled = expiredXostorLicenses.length < xostorLicenses.length
```

Comment:
If you have 2 hosts:
- host A doesn't have a XOSTOR license
- host B has an expired license and a valid license

Then this will set `supportEnabled = true` even though there aren't enough licenses.

```suggestion
            if (expiredXostorLicenses.length === xostorLicenses.length) {
              supportEnabled = false
            }
```